src/music_gui.cpp

```diff
@@ -834,6 +828,31 @@ struct MusicWindow : public Window {
 				break;
 		}
 	}
+
+	virtual void OnMouseDrag(Point pt, int widget)
+	{
+		if (widget == this->dragged_vol_widget && (widget == WID_M_MUSIC_VOL || widget == WID_M_EFFECT_VOL)) {
```

Comment:
Doesn't this extra test for widget mean the mouse cursor has to be on top of the volume widgets? Doesn't this defeat the point?

Reply:
Yes .. this was causing it to not work as well as it should, and not update the volume when the cursor is outside the widget. I've updated the PR.